cmd/vulcan-trivy/vulcan-trivy.go

```diff
@@ -103,65 +158,214 @@ func run(ctx context.Context, target, assetType, optJSON string, state checkstat
+
+		if !opt.ImageEnabled {
+			logger.Warnf("Image scanning is not enabled")
+			return nil
+		}
+
+		// Load required env vars for docker registry authentication.
+		registryEnvDomain := os.Getenv("REGISTRY_DOMAIN")
+		registryEnvUsername := os.Getenv("REGISTRY_USERNAME")
+		registryEnvPassword := os.Getenv("REGISTRY_PASSWORD")
+
+		slashSplit := strings.SplitAfterN(target, "/", 2)
+		if len(slashSplit) <= 1 {
+			logger.Warnf("%s does not have a path", target)
+		}
+		targetSplit := strings.Split(slashSplit[len(slashSplit)-1], ":")
+		if len(targetSplit) != 2 {
+			logger.Warnf("%s does not have a tag", target)
+		}
+
+		registryDomain := strings.Trim(slashSplit[0], "/")
+		// If docker registry equals registryDomain, export trivy credential env vars.
+		if registryDomain == registryEnvDomain {
+			os.Setenv("TRIVY_AUTH_URL", registryEnvDomain)
+			os.Setenv("TRIVY_USERNAME", registryEnvUsername)
+			os.Setenv("TRIVY_PASSWORD", registryEnvPassword)
+		}
+
+		isReachable, err := helpers.IsReachable(target, assetType,
+			helpers.NewDockerCreds(os.Getenv("TRIVY_USERNAME"), os.Getenv("TRIVY_PASSWORD")))
+		if err != nil {
+			logger.Warnf("Can not check asset reachability: %v", err)
+		}
+		if !isReachable {
+			return checkstate.ErrAssetUnreachable
+		}
+
+		results, err := execTrivy(opt, "image", append(trivyArgs, target))
+		if err != nil {
+			return err
+		}
+
+		vuln := report.Vulnerability{
+			// Issue attributes.
+			Summary:     "Outdated Packages in Docker Image",
+			Description: "Vulnerabilities have been found in outdated packages installed in the Docker image.",
+			Recommendations: []string{
+				"Update affected packages to the versions specified in the resources table or newer.",
+			},
+			Details: strings.Join([]string{
+				"Run the following command to obtain the full report in your computer.",
+				"If using a public docker registry:",
+				fmt.Sprintf(`docker run -it --rm aquasec/trivy image %s`, target),
+				"\n",
+				"If using a private docker registry:",
+				fmt.Sprintf(`docker run -it --rm \
+				-e TRIVY_AUTH_URL=https://%s \
+				-e TRIVY_USERNAME=$REGISTRY_USERNAME \
+				-e TRIVY_PASSWORD=$REGISTRY_PASSWORD \
+				aquasec/trivy image %s`, registryEnvDomain, target),
+			}, "\n"),
+			CWEID:  937,
+			Labels: []string{"potential", "docker"},
+			// Finding attributes.
+		}
+		if err = processVulns(results.Results, vuln, "", state); err != nil {
+			logger.Errorf("processing image vuln results: %+v", err)
+		}
+
+		vuln = report.Vulnerability{
+			Summary:       "Secret Leaked in DockerImage",
+			Description:   "A secret has been found stored in the DockerImage. This secret could be retrieved by anyone with access to the image. Test data and false positives can be marked as such.",
+			CWEID:         540,
+			Score:         8.9,
+			ImpactDetails: "Anyone with access to the image could retrieve the leaked secret and use it in the future with malicious intent.",
+			Labels:        []string{"issue"},
```

Comment:
I think that by our standards this should be potential, depending on how confident we are on the results reported.

Reply:
I copied from GitLeaks check. As it will be disabled, perhaps we could keep it as `issue`. Also hey mention that this is "inspired by GitLeaks".

Reply:
If Gitleaks is issue then I agree with leaving it as that consistency 👍

Reply:
I copied from GitLeaks check. As it will be disabled, perhaps we could keep it as `issue`. Also hey mention that this is "inspired by GitLeaks".